helpers/response/create_response.py

```diff
@@ -1,51 +1,61 @@
+            response=True,
+            room_id=kwargs['room_id'],
+            question_id=kwargs['question_id'],
+            created_date=datetime.now())
+        response.question_type = "check"
+        response.save()
+        for item_id in kwargs['missing_items']:
+            missing_item = None
+            room_missing_item = RoomResource.query.filter_by(
+                resource_id=item_id, room_id=kwargs['room_id']).first()
+            if room_missing_item:
+                missing_item = Resource.query.filter_by(
+                    id=item_id).first()
+            if missing_item is None:
+                response.delete()
+                errors.append('Response to question {} was not saved because one of the resources provided was not assigned to the room'.format(kwargs['question_id']))  # noqa
                 return responses, errors
-            rating = Response(
-                rate=kwargs['rate'],
-                room_id=kwargs['room_id'],
-                question_id=kwargs['question_id'],
-                created_date=datetime.now())
-            rating.save()
-            responses.append(rating)
-    if question_type.lower() == 'check':
-        if 'missing_items' in kwargs and kwargs['missing_items']:
-            response = Response(
-                check=True,
-                room_id=kwargs['room_id'],
-                question_id=kwargs['question_id'],
-                created_date=datetime.now())
+            response.missing_resources.append(
+                missing_item
+            )
             response.save()
-            for item_id in kwargs['missing_items']:
-                missing_item = Resource.query.filter_by(
-                    id=item_id, room_id=kwargs['room_id']).first()
-                if not missing_item:
-                    response.delete()
-                    errors.append('Response to question {} was not saved because one of the resource provided does not exist'.format(kwargs['question_id'])) # noqa
-                    return responses, errors
-                response.missing_resources.append(
-                    missing_item
-                )
-                response.save()
             responses.append(response)
-    if question_type.lower() == 'input':
-        if 'text_area' in kwargs and kwargs['text_area']:
-            suggestion = Response(
-                text_area=kwargs['text_area'],
-                room_id=kwargs['room_id'],
-                question_id=kwargs['question_id'],
-                created_date=datetime.now())
-            suggestion.save()
-            responses.append(suggestion)
+    elif question_type.lower() == 'input' and \
+            'text_area' in kwargs and kwargs['text_area']:
+        suggestion = Response(
+            response=kwargs['text_area'],
+            room_id=kwargs['room_id'],
+            question_id=kwargs['question_id'],
+            created_date=datetime.now())
+        suggestion.question_type = 'textarea'
```

Comment:
Good job @Makwe-O, kindly shed some light on something I have noticed. On line 49 we are querying for question type of `input` but on line 56 we are saving it as `textarea` in response. Just for uniformity, could we save both as `input`?